lib/reesmat.gi

```diff
@@ -557,27 +558,31 @@ end);
 # to subsemigroups defined by a generating set, which happen to be
 # simple/0-simple.
 
-InstallMethod(Matrix, "for a Rees matrix semigroup with generators", 
+InstallMethod(MatrixOfReesMatrixSemigroup, "for a Rees matrix semigroup with generators", 
 [IsReesMatrixSubsemigroup and HasGeneratorsOfSemigroup], 
 function(R)
   if not IsReesMatrixSemigroup(R) then 
     return fail;
   fi;
-  return Matrix(ParentAttr(R));
+  return MatrixOfReesMatrixSemigroup(ParentAttr(R));
 end);
 
-InstallMethod(Matrix, "for a Rees 0-matrix semigroup with generators", 
+InstallMethod(MatrixOfReesZeroMatrixSemigroup, "for a Rees 0-matrix semigroup with generators", 
 [IsReesZeroMatrixSubsemigroup and HasGeneratorsOfSemigroup], 
 function(R)
   if not IsReesZeroMatrixSemigroup(R) then 
     return fail;
   fi;
-  return Matrix(ParentAttr(R));
+  return MatrixOfReesZeroMatrixSemigroup(ParentAttr(R));
 end);
 
+# for backwards compatibility
```

Comment:
perhaps this should say "for convenience and backwards compatibility"?

Reply:
Yes, since `Matrix` is documented for these situations.